tools/golang-json-rpc-tests/main.go

```diff
@@ -0,0 +1,412 @@
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block hash %s: %d\n", blockHash.Hex(), txCount)
+}
+
+func testGetBlockTransactionCountByNumber(client *ethclient.Client) {
+    txCount, err := client.PendingTransactionCount(context.Background())
+    if err != nil {
+        log.Fatalf("Failed to get transaction count by block number: %v", err)
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block number: %d\n", txCount)
+}
+
+func testCodeAt(client *ethclient.Client, address common.Address) {
+    code, err := client.CodeAt(context.Background(), address, nil)
+    if err != nil {
+        log.Fatalf("Failed to get code at address: %v", err)
+    }
+    if code == nil {
+        log.Fatalf("Code is nil")
+    }
+    fmt.Printf("Code at address %s: %s\n", address.Hex(), hex.EncodeToString(code)) // Not a SmartContract so empty string expected
+}
+
+func testGetLogs(client *ethclient.Client, address common.Address, topics []common.Hash) {
+    query := ethereum.FilterQuery{
+        Addresses: []common.Address{address},
+        Topics:    [][]common.Hash{topics},
+    }
+
+    logs, err := client.FilterLogs(context.Background(), query)
+    if err != nil {
+        log.Fatalf("Failed to get logs: %v", err)
+    }
+
+    fmt.Printf("Logs for address %s:\n", address.Hex())
+    if len(logs) == 0 {
+        fmt.Printf("No logs found for address %s\n", address.Hex())
+    } else {
+        for _, vLog := range logs { // Not a SmartContract so empty result expected
+            fmt.Printf("Log Block Number: %d\n", vLog.BlockNumber)
+            fmt.Printf("Log Index: %d\n", vLog.Index)
+            fmt.Printf("Log Data: %s\n", hex.EncodeToString(vLog.Data))
+        }
+    }
+}
+
+func testStorageAt(client *ethclient.Client, address common.Address, slot string) {
+    slotHash := common.HexToHash(slot)
+    storageValue, err := client.StorageAt(context.Background(), address, slotHash, nil)
+    if err != nil {
+        log.Fatalf("Failed to get storage at address: %v", err)
+    }
+    if storageValue == nil {
+        log.Fatalf("Storage value is nil")
+    }
+    fmt.Printf("Storage at address %s slot %s: %s\n", address.Hex(), slot, hex.EncodeToString(storageValue))
+}
+
+func testGetTransactionByBlockHashAndIndex(client *ethclient.Client, blockHash common.Hash, index uint) {
+    tx, err := client.TransactionInBlock(context.Background(), blockHash, index)
+    if err != nil {
+        log.Fatalf("Failed to get transaction by block hash and index: %v", err)
+    }
+    fmt.Printf("Transaction in block hash %s at index %d: %s\n", blockHash.Hex(), index, tx.Hash().Hex())
+}
+
+func testGetTransactionByBlockNumberAndIndex(client *ethclient.Client, blockHash common.Hash, index uint) {
+    tx, err := client.TransactionInBlock(context.Background(), blockHash, index)
```

Comment:
Looks like it's supposed to be blockNum here, instead of blockHash?

Reply:
Yes, and Go ethereum client does not seem to have this call to: eth_getTransactionByBlockNumber implemented at all. So I've just removed this test case from this PR.